### DISCUSSION TITLE
Changes to live environments

### DISCUSSION THREAD

**Discussion by ralf-berger** [ude-soco/CourseMapper-webserver#1288] [Announcements]
Initiated by https://github.com/ude-soco/CourseMapper-webserver/issues/1275, the [production](http://coursemapper.de/) environment will switch from running the *latest patch release* to running *edge* (i.e. the latest automatic build from the main branch). The edge branch preview is therefore obsolete.

New (long running integration) branches will get their own permanent preview environment:
* Branch [dev](https://br-dev.coursemapper.de/)

#### Notes:
* Tagged application releases are not going to be deployed to any live environments for now.
* It will become harder to tell what version is running in production. You may want to space merges to `main` further apart, keep an eye on GitHub Action runs and compare updates times of container images and deployed instances.

**Comment by rawaa123** [comment]
> * It will become harder to tell what version is running in production. You may want to space merges to `main` further apart, keep an eye on GitHub Action runs and compare updates times of container images and deployed instances.

1- Could you please clarrify me more about the the seond point. 

2- In the discussion [here](https://github.com/ude-soco/CourseMapper-webserver/discussions/1289), you mentioned that the new updates include the use of #hash instead of Tags. Does this mean we will no longer have releases and their details, as we did in Release v4.1.0? [Release v4.1.0 Link](https://github.com/ude-soco/CourseMapper-webserver/releases/tag/v4.1.0)

**Reply by ralf-berger** [reply]
> Could you please clarrify me more about the the seond point.

Because of the switch from deploying releases (with human-readable version number tags) to deploying any untagged commit (to `main`), it becomes harder to tell what's running in production—e.g. which changes the currently running container images include and if they originate from the same revision. Checking errors in GitHub Action runs, making sure that builds do not overlap and comparing build and update times might make it easier to keep an overview.

> Does this mean we will no longer have releases and their details, as we did in Release v4.1.0?

It is still possible to occasionally create releases, but (a) it is a bit pointless since they are not deployed anywhere and (b) the changelogs will be based on the pull requests from `dev` to `main`, so those must include meaningful titles representing the changes included.
